parachain/pallets/control/src/lib.rs

```diff
@@ -158,6 +165,91 @@ pub mod pallet {
 			Self::deposit_event(Event::<T>::CreateAgent { location: Box::new(location), agent_id });
 			Ok(())
 		}
+
+		/// Sends a message to the Gateway contract to create a new channel for `origin`
+		///
+		/// - `origin`: Must be `MultiLocation`
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::create_agent())]
+		pub fn create_channel(origin: OriginFor<T>) -> DispatchResult {
+			let location: MultiLocation = T::CreateAgentOrigin::ensure_origin(origin)?;
+
+			let (agent_id, para_id, location) = Self::convert_location(location)?;
+
+			if !Agents::<T>::contains_key(agent_id) {
+				return Err(Error::<T>::AgentNotExist.into())
+			}
+
+			let message = Message {
+				origin: T::OwnParaId::get(),
+				command: Command::CreateChannel { agent_id, para_id },
+			};
+			Self::submit_outbound(message)?;
+
+			Self::deposit_event(Event::<T>::CreateChannel {
+				location: Box::new(location),
+				para_id,
+				agent_id,
+			});
+
+			Ok(())
+		}
+
+		/// Sends a message to the Gateway contract to update channel
+		///
+		/// - `origin`: Must be `MultiLocation`
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::create_agent())]
+		pub fn update_channel(
+			origin: OriginFor<T>,
+			mode: OperatingMode,
+			fee: u128,
+			reward: u128,
+		) -> DispatchResult {
+			let location: MultiLocation = T::CreateAgentOrigin::ensure_origin(origin)?;
```

Comment:
Maybe we should rename this to `AgentOrigin` as its used for more than just create agent.
```suggestion
			let location: MultiLocation = T::AgentOrigin::ensure_origin(origin)?;
```

Reply:
I would prefer to rename as `ControlOrigin` as now applies also to `create_channel`.

Reply:
I like that name.